tests/js/spec/views/organizationGroupDetails/groupSimilarIssues.spec.jsx

```diff
@@ -41,77 +44,69 @@ describe('Issues Similar View', function () {
     });
   });
 
+  afterEach(() => {
+    MockApiClient.clearMockResponses();
+    jest.clearAllMocks();
+  });
+
   it('renders initially with loading component', function () {
-    const component = mountWithTheme(
+    render(
       <GroupSimilarIssues
         project={project}
         params={{orgId: 'org-slug', groupId: 'group-id'}}
         location={{}}
       />,
-      routerContext
+      {context: routerContext}
     );
 
-    expect(component).toSnapshot();
+    expect(screen.getByTestId('loading-indicator')).toBeInTheDocument();
   });
 
   it('renders with mocked data', async function () {
-    const wrapper = mountWithTheme(
+    const wrapper = render(
       <GroupSimilarIssues
         project={project}
         query=""
         params={{orgId: 'org-slug', projectId: 'project-slug', groupId: 'group-id'}}
         location={{}}
       />,
-      routerContext
+      {context: routerContext}
     );
 
-    await tick();
-    await tick();
-    wrapper.update();
-    expect(mock).toHaveBeenCalled();
-    expect(wrapper.find('SimilarStackTrace')).toSnapshot();
+    await waitFor(() => expect(mock).toHaveBeenCalled());
+    expect(wrapper.container).toSnapshot();
   });
 
   it('can merge and redirect to new parent', async function () {
-    const wrapper = mountWithTheme(
-      <GroupSimilarIssues
-        project={project}
-        params={{orgId: 'org-slug', projectId: 'project-slug', groupId: 'group-id'}}
-        location={{}}
-      />,
-      routerContext
-    );
     const merge = MockApiClient.addMockResponse({
       method: 'PUT',
       url: '/projects/org-slug/project-slug/issues/',
       body: {
         merge: {children: ['123'], parent: '321'},
       },
     });
-
-    await tick();
-    await tick();
-    wrapper.update();
-
-    wrapper.find('[data-test-id="similar-item-row"]').first().simulate('click');
-
-    await tick();
-    wrapper.update();
-
-    wrapper.find('[data-test-id="merge"] button').simulate('click');
-
-    const modal = await mountGlobalModal();
-    modal.find('Button[data-test-id="confirm-button"]').simulate('click');
-
-    await tick();
-    wrapper.update();
-
-    expect(merge).toHaveBeenCalledWith(
-      '/projects/org-slug/project-slug/issues/',
-      expect.objectContaining({
-        data: {merge: 1},
-      })
+    render(
+      <GroupSimilarIssues
+        project={project}
+        params={{orgId: 'org-slug', projectId: 'project-slug', groupId: 'group-id'}}
+        location={{}}
```

Comment:
```suggestion
   project={project}
        params={{orgId: 'org-slug', groupId: 'group-id'}}
        location={router.location}
        router={router}
        routeParams={router.params}
        routes={router.routes}
        route={{}}
```

Reply:
for these suggested changes is necessary to have `const router = TestStubs.router();`